client/src/questions-n-answers/qNa.jsx

```diff
@@ -57,20 +57,37 @@ class QuestionsNAnswers extends React.Component {
     this.updateAnswers = this.updateAnswers.bind(this)
     this.addAnswer = this.addAnswer.bind(this)
     this.addAnswerOnClick = this.addAnswerOnClick.bind(this)
-
+    this.addToReported = this.addToReported.bind(this)
 
   }
   componentDidMount() {
-    console.log(this.props)
-    let copy = this.props.data.slice()
-    let sortedData= this.filterAnswersNQuestions(copy)
-    let showButton = this.helper().showMoreAnsweredQuestions(sortedData)
-    console.log(sortedData)
-    this.setState({
-      questions: sortedData[0],
-      answers: sortedData[1],
-      showQuestionButton: showButton
-    })
+    getReportedAns()
+     .then(data => {
+       let answerIds = data.data
+       let copy = this.props.data.slice()
+       let sortedData= this.filterAnswersNQuestions(copy)
+       let showButton = this.helper().showMoreAnsweredQuestions(sortedData)
```

Comment:
I would worry here from errors in my component that bugs could stem from the helper func on line 68 not completing before the code on 69 runs. As longer as the filter is fast, I'd imagine it could be fine-- but if it takes a while, you might get undefined as the value for sortedData on 69